### PR TITLE
[v0.12.x] Add transaction balance fix

### PR DESCRIPTION
**Pull request by zsluedem**
## Overview

Duplicated PRs from dev branch to `v0.12.x` release.
- #3624
- #3626

add https://github.com/rchain/rchain/pull/3624/commits and https://github.com/rchain/rchain/pull/3626/commits into v0.12

The old pr is merged into 0.13.0 which the radix history is already changed.

We would need that in the future for hard fork 2 to take a snapshot.

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by zsluedem** [comment]
bors try

**Comment by tgrospic** [comment]
The branch which follow current main net should be `feature/v0.12` on which we will create tag for the release `v0.12.6`.

`feature/v0.12` branch should then be merged to `feature/v0.12-rpc` for RPC release.

Please also add PRs to GitHub core project.

File: node/src/main/scala/coop/rchain/node/revvaultexport/mainnet1/reporting/MergeBalanceMain.scala [hunk 1]
```diff
@@ -181,23 +181,25 @@ object MergeBalanceMain {
...
+                                             postStateHash
+                                           )
+                                         else 0L.pure[Task]
                                adjustAccount = account.copy(
                                  adjustedStateBalance = balance
                                )
                                _ <- Log[Task]
                                      .info(
-                                       s"Should Before adjusted ${account} after ${adjustAccount}"
+                                       s"Should Before adjusted after ${adjustAccount}"
```
Comment (tgrospic) [inline node/src/main/scala/coop/rchain/node/revvaultexport/mainnet1/reporting/MergeBalanceMain.scala]: Please make this log message more clear. What it means _before adjusted after_?

**Review by tgrospic — APPROVED** [review]
LGTM

**Pull request merged**